src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,114 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] array;
+    private int size = 0;
+
+    public ArrayList() {
+        this.array = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == array.length) {
+            capacity();
+        }
+        array[size] = value;
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > size) {
+            throw new ArrayListIndexOutOfBoundsException("Out of bound");
+        }
+        if (size == array.length) {
+            capacity();
+        }
+        if (index == size) {
+            array[size] = value;
+        } else {
+            System.arraycopy(array, index, array, index + 1, size - index);
+            array[index] = value;
+        }
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
 
+        T[] newArray = (T[]) new Object[list.size()];
```

Comment:
Creating a new array `newArray` is unnecessary here. You can directly add elements from the `list` to the current array using the `add` method.